### PR TITLE
[feat] Add Webb Chains Info crate

### LINKED ISSUE #412: [TASK] Remove the hardcoded `NativeToken` names while fetching prices
### Overview

Instead of Hard-coding chain native token names in the relayer, we should instead use something like 

```
https://chainid.network/chains.json
```

Which contains a big list of networks/chains, each has a `chainId`, a `name` and `nativeCurrency` with its `symbol` and decimals too!

We should make use of that and bundle this file with the relayer binary, no need to fetch it every time.

### Coingecko-Specific Token Ids

Coingecko uses it's own set of token ids, different than just using the token symbol or ticker, hence, the generated chains.json file should include somehow the corresponding coingecko token id by query the 
```
https://api.coingecko.com/api/v3/coins/list?include_platform=false
```
endpoint which contains the `id`, `symbol` and `name`.

The result `chains.json` file should include the minimum information needed by the relayer, for now the `ChainInfo` struct may look like the following:

```rust
struct ChainInfo {
    pub name: String,
    pub short_name: String,
    pub chain_id: u32,
    pub native_currency: ChainNativeCurrency,
}

struct ChainNativeCurrency {
    pub name: String,
    pub symbol: String,
    pub decimals: u8,
   pub coingecko_token_id: Option<String>,
}
```

### Task Checklist

- [x] Add `chains.json` in the repos fixtures
- [x] Include the Chains in the relayer binary (e.g by using `include_str!` macro)
- [x] Create a simple representation of `Chain` that contains the min info we need from that JSON file
- [x] Parse and store the chains as a map between `TypedChainId` to `ChainInfo` and add it to the relayer context.
- [x] Make use of the chains to fetch the native token name and the decimals while calculating the fees.

### PR DESCRIPTION
**Pull request by shekohex**
## Summary of changes
- Adds a new crate `webb-chains-info` that contains the chain information of the supported chains by the relayer.
- bundles the `chains.json` from `https://chainid.network/chains.json`
- bundles the `coingecko_coins_list.json` from `https://api.coingecko.com/api/v3/coins/list?include_platform=false`
- added a simple file to control which chains are supported internally by the relayer (see `supported_chains.toml` file)
- updated the fees endpoint to not relay on the hardcoded coingecko-specific coin ids and instead use the new package.
- updated the coingecko price oracle backend to use the new package to map the token symbols to coingecko's coin ids.

### Reference issue to close (if applicable)
- Closes #412 

### How to add a new chain?

Just add the Chain Id to the list of chain ids inside the `supported_chains.toml` file and rebuild the relayer. The generated file is not included in the git history, but it gets generated during the build of the `webb-chains-info` crate.

Here is an example of the generated file:

<details>
  <summary>Click to view</summary>
  
  ```rs
  mod chains {
    //! Chains Information
    //!
    //! This file is generated by the `build.rs` script.
    //! Do not edit this file manually.
    //!
    //! This file contains the information about the chains that are supported by the relayer.
    //! The information is used to generate the `chains.rs` file and could be used by
    //! the relayer to get the information about the chains.
    /// The Chain Information.
    pub struct ChainInfo {
        /// Chain Identifier.
        pub chain_id: u64,
        /// Chain Name.
        pub name: &'static str,
        /// Chain Short Name, usually the ticker.
        pub short_name: &'static str,
        /// Chain Native Currency Information.
        pub native_currency: CurrencyInfo,
    }
    /// The Currency Information.
    pub struct CurrencyInfo {
        /// Currency Name.
        pub name: &'static str,
        /// Currency Symbol.
        pub symbol: &'static str,
        /// Currency Decimals.
        pub decimals: u8,
        /// Coingecko's Coin Identifier.
        ///
        /// This is `None` if the chain is not supported by Coingecko.
        pub coingecko_coin_id: Option<&'static str>,
    }
    pub type ChainsInfo = [(u64, ChainInfo); 6usize];
    /// List of all the supported chains.
    pub const CHAINS_INFO: ChainsInfo = [
        (
            5u64,
            ChainInfo {
                chain_id: 5u64,
                name: "Goerli",
                short_name: "gor",
                native_currency: CurrencyInfo {
                    name: "Goerli Ether",
                    symbol: "ETH",
                    decimals: 18u8,
                    coingecko_coin_id: Some("ethereum"),
                },
            },
        ),
        (
            1287u64,
            ChainInfo {
                chain_id: 1287u64,
                name: "Moonbase Alpha",
                short_name: "mbase",
                native_currency: CurrencyInfo {
                    name: "Dev",
                    symbol: "DEV",
                    decimals: 18u8,
                    coingecko_coin_id: Some("moonbeam"),
                },
            },
        ),
        (
            80001u64,
            ChainInfo {
                chain_id: 80001u64,
                name: "Mumbai",
                short_name: "maticmum",
                native_currency: CurrencyInfo {
                    name: "MATIC",
                    symbol: "MATIC",
                    decimals: 18u8,
                    coingecko_coin_id: Some("matic-network"),
                },
            },
        ),
        (
            421613u64,
            ChainInfo {
                chain_id: 421613u64,
                name: "Arbitrum Goerli",
                short_name: "arb-goerli",
                native_currency: CurrencyInfo {
                    name: "Arbitrum Goerli Ether",
                    symbol: "AGOR",
                    decimals: 18u8,
                    coingecko_coin_id: Some("arbitrum"),
                },
            },
        ),
        (
            534353u64,
            ChainInfo {
                chain_id: 534353u64,
                name: "Scroll Alpha Testnet",
                short_name: "scr-alpha",
                native_currency: CurrencyInfo {
                    name: "Ether",
                    symbol: "ETH",
                    decimals: 18u8,
                    coingecko_coin_id: Some("ethereum"),
                },
            },
        ),
        (
            11155111u64,
            ChainInfo {
                chain_id: 11155111u64,
                name: "Sepolia",
                short_name: "sep",
                native_currency: CurrencyInfo {
                    name: "Sepolia Ether",
                    symbol: "ETH",
                    decimals: 18u8,
                    coingecko_coin_id: Some("ethereum"),
                },
            },
        ),
    ];
}

  ```
  
</details>

-----
### Code Checklist 

- [x] Tested
- [x] Documented

### CODE REVIEW

File: crates/chains-info/supported_chains.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+chain-ids = [
+    5, # Goerli Testnet
+    80001, # Polygon Mumbai Testnet
+    421613, # Arbitrum Goerli Testnet
+    534353, # Scroll Alpha Testnet
+    11155111, # Sepolia Testnet
+    1287, # Moonbase Alpha
+]
```
Comment (shekohex) [inline crates/chains-info/supported_chains.toml:8]: @salman01zp @dutterbutter What EVM Based chains we should support initially?

Comment (drewstone) [reply crates/chains-info/supported_chains.toml:8]: All the ones for which we have testnets

File: crates/chains-info/supported_chains.toml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Since the Moonbase Alpha network is not yet listed on CoinGecko.
+# so we use the mainnet coin id instead.
+[overrides.1287]
+coingecko-coin-id = "moonbeam"
+
+# Override the coingecko coin Id for Arbitrum Goerli Testnet
+# Since the Arbitrum Goerli Testnet is not yet listed on CoinGecko.
+# so we use the mainnet coin id instead.
+[overrides.421613]
+coingecko-coin-id = "arbitrum"
```
Comment (drewstone) [inline crates/chains-info/supported_chains.toml:20]: If this is testnet can we name it such? Since there is both a testnet and mainnet

**Review by drewstone — COMMENTED** [review]
LG

File: crates/chains-info/supported_chains.toml (hunk 1 above)
Comment (shekohex) [reply crates/chains-info/supported_chains.toml]: Done.

File: crates/chains-info/supported_chains.toml (hunk 2 above)
Comment (shekohex) [reply crates/chains-info/supported_chains.toml]: Updated

**Pull request merged**